File: src/main/java/com/negosud/api/service/FournisseurService.java
package com.negosud.api.service;

import com.negosud.api.dto.request.FournisseurRequestDto;
import com.negosud.api.dto.response.ArticleResponseDto;
import com.negosud.api.dto.response.FournisseurResponseDto;

import java.util.List;

public interface FournisseurService {

    List<FournisseurResponseDto> findAll();

    FournisseurResponseDto findById(Long id);

    List<ArticleResponseDto> findArticlesByFournisseur(Long id);

    FournisseurResponseDto create(FournisseurRequestDto dto);

    FournisseurResponseDto update(Long id, FournisseurRequestDto dto);

    void delete(Long id);
}

File: front/src/services/fournisseurService.js
import api from './api';

const BASE = '/fournisseurs';

export const fournisseurService = {
  findAll: () => api.get(BASE),
  findById: (id) => api.get(`${BASE}/${id}`),
  findArticles: (id) => api.get(`${BASE}/${id}/articles`),
  create: (data) => api.post(BASE, data),
  update: (id, data) => api.put(`${BASE}/${id}`, data),
  delete: (id) => api.delete(`${BASE}/${id}`),
};
